src/sentry/static/sentry/app/components/events/interfaces/frameRegisters.jsx

```diff
@@ -0,0 +1,139 @@
+import {Flex, Box} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import {defined, objectToArray} from 'app/utils';
+
+const REGISTER_VIEWS = [t('Hexadecimal'), t('Numeric')];
+
+export class RegisterValue extends React.Component {
+  static propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      view: 0,
+    };
+  }
+
+  toggleView = () => {
+    this.setState({
+      view: (this.state.view + 1) % REGISTER_VIEWS.length,
+    });
+  };
+
+  formatValue = value => {
+    try {
+      let parsed = typeof value === 'string' ? parseInt(value, 16) : value;
+      if (isNaN(parsed)) return value;
+
+      switch (this.state.view) {
+        case 1:
+          return String(parsed);
+        case 0:
+        default:
+          return '0x' + ('0000000000000000' + parsed.toString(16)).substr(-16);
```

Comment:
nit: can we use string templates here?

Also, why aren't we using `left-pad` here? :trollface:

Reply:
Always a fan of these micro libraries 🤣 Not a fan of adding for this only use, but if it comes up more often I'm happy to refactor.